### PR TITLE
[FIX] mail: avoid hiding non-text messages that are not empty

### PR DESCRIPTION
**Pull request by seb-odoo**
task-2447695

### CODE REVIEW

File: addons/mail/static/src/models/message/message.js [hunk 1]
```diff
@@ -420,11 +420,33 @@ function factory(dependencies) {
...
+                    [
+                        '',
+                        '<p></p>',
+                        '<p><br></p>',
+                        '<p><br/></p>',
+                    ].includes(this.body.replace(/\s/g, ''))
+                ) &&
                 this.attachments.length === 0 &&
                 this.tracking_value_ids.length === 0 &&
                 !this.subtype_description
```
Comment (alexkuhn) [inline addons/mail/static/src/models/message/message.js:452]: I don't like long conditional statements with `||`.
Feels way too academic and hard to read, especially when it's incorporated in a longer conditional statement containing `&&`.

I prefer you define a boolean and local variable for what is stated as empty body.
JS sucks a bit when you want to scope computation of a local variable, and also when pattern matching features is only old `switch/case`... But the idea is so:
```js
const isBodyEmpty = (() => {
    if (!this.body) {
        return true;
    }
    const bodyWithoutSpaces = this.body.replace(/\s/g, '');
    switch (bodyWithoutSpaces) {
        case '':
        case '<p></p>':
        case '<p><br></p>':
        case '<p><br/></p>':
            return true;
        default:
            return false;
    }
})();

return (
    isBodyEmpty &&
    [...]
);
```

It's longer, but I feel it's less cryptic overall and easier to understand at a glance, while being also easier to maintain. What do you think?

_(just to be clear, this comment is really a very small deal and more random ideas)_

Comment (seb-odoo) [reply addons/mail/static/src/models/message/message.js:452]: I understand what you mean and I agree (I don't like conditions with `or` in general and especially mixed with `and`) but I feel like introducing a function for that doesn't make it more readable.

I would prefer in that case just make 1 variable for the `or` part (and depending on what we need, could even be its own field) and then use that inside the `and` part.

Comment (seb-odoo) [reply addons/mail/static/src/models/message/message.js]: Done the split with just a variable. It's a bit better. If we really want a separate function, I would make a new field instead.

**Review by alexkuhn — APPROVED** [review]
LGTM

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**